hasktorch/src/Torch/Typed/Native.hs

```diff
@@ -2943,9 +2943,72 @@ qZeroPoint input = unsafePerformIO $ cast1 ATen.q_zero_point_t input
 -- combinations :: Tensor device dtype shape -> Int -> Bool -> Tensor device dtype shape
 -- combinations _input _r _with_replacement = unsafePerformIO $ (cast3 ATen.combinations_tlb) _input _r _with_replacement
 
--- | lstmCell
--- >>> dtype &&& shape $ fst $ lstmCell (ones :: CPUTensor 'D.Float '[12,2]) (ones :: CPUTensor 'D.Float '[12,3]) (ones :: CPUTensor 'D.Float '[12]) (ones :: CPUTensor 'D.Float '[12]) ((ones :: CPUTensor 'D.Float '[2,3]), (ones :: CPUTensor 'D.Float '[2,3])) (ones :: CPUTensor 'D.Float '[2,2])
--- (Float,[2,3])
+-- | lstm
+-- Parameters for this ATen function are non-trivially provided.  See the
+-- `Typed.NN.LSTM` module for doctests.
+--
+lstm
+  :: forall
+       numLayers
+       numDirections
```

Comment:
there are only two valid choices, 1 and 2, right? wouldn't it be better to use the ADT you defined in the other module here?

Reply:
It depends on how you want the modules structured.  I saw the `Native` module as an unsafe, internal module.  You could beef up the safety but I think it would add a fair amount of code in that sourcefile when all is said and done.  

I had a lot of stuff along side the `lstm` function thinking that it would also be used in the other recurrent cells, but it got cluttered and I thought it would be better to just focus on the `lstm` before factoring our the shared machinery.

Reply:
initially, the `Typed.Native` module was supposed to be automatically generated and a thin wrapper, but, after manually implementing many ops, it turns out that it won't be possible to automatically generate the constraints for all ops in there for quite some time. `Typed.Native` has just become a place where we implemented type-safe ops and also put auxiliary type families and classes. Therefore, I think it does make sense to put helpers in there. If it is a generic helper, consider also `Typed.Aux`.

Reply:
addressed

Reply:
thank you!